File: gto-support-eventbus/src/main/java/org/ccci/gto/android/common/eventbus/TimberLogger.java
package org.ccci.gto.android.common.eventbus;

import org.greenrobot.eventbus.Logger;

import java.util.logging.Level;

import timber.log.Timber;

public final class TimberLogger extends Logger.AndroidLogger {
    private static final String TAG = "EventBus";

    public TimberLogger() {
        super(TAG);
    }

    @Override
    public void log(final Level level, final String msg) {
        Timber.tag(TAG).log(mapLevel(level), msg);
    }

    @Override
    public void log(final Level level, final String msg, final Throwable th) {
        Timber.tag(TAG).log(mapLevel(level), th, msg);
    }
}

File: gto-support-eventbus/src/main/java/org/ccci/gto/android/common/eventbus/TimberLogger.kt
package org.ccci.gto.android.common.eventbus

import android.util.Log
import androidx.annotation.VisibleForTesting
import java.util.logging.Level
import org.greenrobot.eventbus.Logger
import timber.log.Timber

private const val TAG = "EventBus"

object TimberLogger : Logger {
    override fun log(level: Level, msg: String?) = Timber.tag(TAG).log(mapLevel(level), msg)
    override fun log(level: Level, msg: String?, th: Throwable?) = Timber.tag(TAG).log(mapLevel(level), th, msg)

    @VisibleForTesting
    internal fun mapLevel(level: Level): Int {
        val value = level.intValue()
        return when {
            value >= Level.SEVERE.intValue() -> Log.ERROR
            value >= Level.WARNING.intValue() -> Log.WARN
            value >= Level.INFO.intValue() -> Log.INFO
            value >= Level.CONFIG.intValue() -> Log.DEBUG
            else -> Log.VERBOSE
        }
    }
}
